### ISSUE TITLE
issue with textinput keyboard_suggestions or InputType on android when using backspace

### ISSUE THREAD

**Issue by Dro12u** [kivy/kivy#8625]
**Software Versions**
* Python: 3.10.4
* OS: windows10/ ubuntu 22.04
* Kivy: v2.3.0
* tested on the latest version of android

**Describe the bug**
basically what happens is when i have inputtype and/or keyboard_suggestions set, the backspace does some weird behavior shown in the video attached. this doesn't happen on windows only on phone/tablet 

**Expected behavior**
it would be nice to have the keyboard_suggestions feature working properly for my users.

**To Reproduce**
issue happens when i have my textinput set like this:
         self.text_input = TextInput(input_type='text',hint_text=f"{self.instructions_message}", keyboard_suggestions=True, multiline=True,  size_hint=(1, 0.8) )
or this:
         self.text_input = TextInput(input_type='text', hint_text=f"{self.instructions_message}", multiline=True,  size_hint=(1, 0.8) )

**Code and Logs and screenshots**

https://github.com/kivy/kivy/assets/149750354/7abec06b-8510-4ae6-bba2-219dd1b7cf38

```

**Additional context**
tried to download latest version of sdl 
tried to use develop instead of master for p4a.branch in my .spec file 
the only way to have this behavior fixed is by fully removing the inputtype and the keyboard_suggestions from the textinput

**Comment by misl6** [comment]
I do not think we have a specific one, even if this issue already showed up somewhere, so I will keep this open.
However, you may find it interesting to check out [this recap](https://github.com/kivy/kivy/issues/8575).
